src/main/java/org/embulk/input/marketo/MarketoServiceImpl.java

```diff
@@ -0,0 +1,183 @@
+package org.embulk.input.marketo;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.Iterables;
+import com.google.common.io.ByteStreams;
+import com.google.common.io.Files;
+import org.embulk.input.marketo.rest.MarketoRestClient;
+import org.embulk.input.marketo.rest.RecordPagingIterable;
+import org.embulk.spi.Column;
+import org.embulk.spi.DataException;
+import org.embulk.spi.Exec;
+import org.embulk.spi.TempFileSpace;
+import org.embulk.spi.type.Types;
+import org.slf4j.Logger;
+
+import javax.annotation.Nullable;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Created by tai.khuu on 9/6/17.
+ */
+public class MarketoServiceImpl implements MarketoService
+{
+    private static final Logger LOGGER = Exec.getLogger(MarketoServiceImpl.class);
+
+    private static final String DEFAULT_FILE_FORMAT = "csv";
+
+    private static final String LIST_ID_COLUMN_NAME = "listId";
+
+    private static final String PROGRAM_ID_COLUMN_NAME = "programId";
+
+    private MarketoRestClient marketoRestClient;
+
+    public MarketoServiceImpl(MarketoRestClient marketoRestClient)
+    {
+        this.marketoRestClient = marketoRestClient;
+    }
+
+    @Override
+    public File extractLead(Date startTime, Date endTime, List<String> extractFields, int pollingTimeIntervalSecond, int bulkJobTimeoutSecond)
+    {
+        String exportID = marketoRestClient.createLeadBulkExtract(startTime, endTime, extractFields);
+        marketoRestClient.startLeadBulkExtract(exportID);
+        try {
+            marketoRestClient.waitLeadExportJobComplete(exportID, pollingTimeIntervalSecond, bulkJobTimeoutSecond);
+        }
+        catch (InterruptedException e) {
+            LOGGER.error("Exception when waiting for export job id: {}", exportID, e);
+            throw new DataException("Error when wait for bulk extract");
+        }
+        InputStream leadBulkExtractResult = marketoRestClient.getLeadBulkExtractResult(exportID);
+        final File tempDir = Files.createTempDir();
+        TempFileSpace tempFiles = new TempFileSpace(tempDir);
```

Comment:
This `TempFileSpace` needed to be cleaned up.

Reply:
Fixed by using Exec.getTemspace instead